test/discrepancies/Nonce.js

```diff
@@ -299,12 +342,586 @@ describe('@discrepancies - Nonce Test Suite', async () => {
     const snBefore = await getServicesNonce(signers[0].address);
     const mnBefore = await getMirrorNodeNonce(signers[0].address);
 
-    const tx = await internalCalleeContract.factorySample({gasLimit: 500_000});
+    const tx = await internalCalleeContract.factorySample({ gasLimit: 500_000 });
     await tx.wait();
 
     const snAfter = await getServicesNonce(signers[0].address);
     const mnAfter = await getMirrorNodeNonce(signers[0].address);
 
     expectIncrementedNonce(snBefore, mnBefore, snAfter, mnAfter);
   });
+
+  //NONCE-017
+  it('should update nonce after successful ERC20 token call', async function () {
+
+    const signers = await ethers.getSigners();
+    const amount = 200;
+
+    const snBefore = await getServicesNonce(signers[0].address);
+    const mnBefore = await getMirrorNodeNonce(signers[0].address);
+
+    try {
+      const tx = await erc20Contract
+        .connect(signers[0])
+        .transfer(
+          tokenAddress,
+          signers[1].address,
+          amount,
+          Constants.GAS_LIMIT_1_000_000
+        );
+      await tx.wait();
+    } catch (e) {
+      expect(e).to.exist;
+      expect(e.code).to.eq(Constants.CALL_EXCEPTION);
+    }
+
+    const snAfter = await getServicesNonce(signers[0].address);
+    const mnAfter = await getMirrorNodeNonce(signers[0].address);
+
+    expectIncrementedNonce(snBefore, mnBefore, snAfter, mnAfter);
+  })
+
+  //NONCE-017
+  it('should update nonce after successful ERC721 token call', async function () {
+    erc721Contract = await Utils.deployERC721Contract();
+    tokenAddress = await Utils.createNonFungibleToken(
+      tokenCreateContract,
+      signers[0].address
+    );
```

Comment:
Let's avoid overriding variables which are reused in multiple tests - this creates dependencies between the tests and depending on their execution order the variable might hold a different value. This reduces the readability of code and makes it hard to trace what is happening without knowing the order of the tests or without going into debug mode.

If we are going to reuse a variable let's assign it only in one place (e.g., in a `before()` method or, where possible, directly turn it into a `const`.)

Please check for other such occurrences and try to fix them 🙂. Thank you!

Reply:
Unneeded override is removed

Reply:
right